### PR TITLE
Habana Synapseai v1.2.0 dockerfiles

### PR DESCRIPTION
**Pull request by omrialmog**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Update docker image path to 1.2.0 release docker
Added 1.10.0 pytorch
Added 2.7.0 tensorflow

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by tejaschumbalkar — COMMENTED** [review]
@waytrue17 Can you address the comments from #1551

**Comment by waytrue17** [comment]
> @waytrue17 Can you address the comments from #1551
Done

File: habana/pytorch/training/docker/1.10/py3/synapseai1.2.0/Dockerfile.hpu [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    psutil \
+    Pillow \
+    scipy \
+    pybind11 \
+    mpi4py==3.0.3 \
+    cmake==3.18.2.post1 \
+    torchnet \
+    "cryptography>3.2" \
+    werkzeug \
+ && pip uninstall -y dataclasses typing \
```
Comment (tejaschumbalkar) [inline habana/pytorch/training/docker/1.10/py3/synapseai1.2.0/Dockerfile.hpu:56]: Lets remove this step

Comment (omrialmog) [reply habana/pytorch/training/docker/1.10/py3/synapseai1.2.0/Dockerfile.hpu:56]: The uninstall step specifically? I think this was a workaround you had placed for previous release, we can remove the uninstall

Comment (omrialmog) [reply habana/pytorch/training/docker/1.10/py3/synapseai1.2.0/Dockerfile.hpu:56]: Done

File: habana/tensorflow/training/docker/2.7/py3/synapseai1.2.0/Dockerfile.hpu [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+RUN apt-get update && apt-get -y install cmake protobuf-compiler
+
+RUN ${PIP} uninstall -y tensorflow-cpu \
+ && ${PIP} install --no-cache-dir -U \
+    numpy==1.19.1 \
+    scipy==1.5.2 \
+    scikit-learn==0.23 \
+    pandas==1.1 \
+    "Pillow>=8.3,<8.4" \
```
Comment (tejaschumbalkar) [inline habana/tensorflow/training/docker/2.7/py3/synapseai1.2.0/Dockerfile.hpu:97]: Can we unpin the pillow version?

Comment (omrialmog) [reply habana/tensorflow/training/docker/2.7/py3/synapseai1.2.0/Dockerfile.hpu]: Done

**Review by waytrue17 — APPROVED** [review]
LGTM, thanks!

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**